strongbox.go

```diff
@@ -110,19 +112,27 @@ func genKey(desc string) {
 		log.Fatal(err)
 	}
 
-	pub, priv, err := box.GenerateKey(rand.Reader)
+	priv := make([]byte, 32)
+	_, err = rand.Read(priv)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	keyRing.AddKey(desc, *pub, *priv)
+	pub := hash(priv)
+
+	keyRing.AddKey(desc, pub, priv)
 
 	err = keyRing.Save()
 	if err != nil {
 		log.Fatal(err)
 	}
 }
 
+func hash(in []byte) []byte {
+	h := crypto.SHA256.New()
+	return h.Sum(nil)
```

Comment:
actually, no. it appends the hash to the value passed and returns the result.  There's another major bug though. I'll push an update.